install/kubernetes/cilium/templates/cilium-configmap.yaml

```
@@ -946,3 +946,15 @@ data:
 {{- end }}
 
 {{- end }}
+
+---
+{{- if and .Values.ipMasqAgent .Values.ipMasqAgent.enabled .Values.ipMasqAgent.config }}
```

Go template will evaluate all operands of `and` depending on the version (used to compile Helm), so checking for `.Values.ipMasqAgent` won't help in that case (e.g. if it's set to `~`). At that point, I would suggest to rely on `.Values.ipMasqAgent.enabled` only.

```suggestion
{{- if and .Values.ipMasqAgent.enabled .Values.ipMasqAgent.config }}
```

---

Thanks for your point! Done.